pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1316,10 +1316,55 @@ func doActivateTail(ctx *domainContext, status *types.DomainStatus,
 			status.Key())
 	}
 	status.Activated = true
+	setupVlans(status.VifList)
 	log.Functionf("doActivateTail(%v) done for %s",
 		status.UUIDandVersion, status.DisplayName)
 }
 
+// VLAN filtering could have been enabled on the bridge immediately after
+// it's creation in zedrouter itself. For some strange reason netlink
+// throws back error saying that the device is busy is vlan filtering is enabled
+// immediately creation.
+// We are either keep retrying in a loop in zedrouter or enable it when needed
+// in domainmgr
+func enableVlanFiltering(bridgeName string) {
+	bridge, err := netlink.LinkByName(bridgeName)
+	if err != nil {
+		log.Errorf("isVlanFilteringSet: LinkByName failed for %s: %s", bridgeName, err)
+		return
+	}
+	if !*bridge.(*netlink.Bridge).VlanFiltering {
+		// Enable VLAN filtering on bridge
+		if err := netlink.BridgeSetVlanFiltering(bridge, true); err != nil {
+			errStr := fmt.Sprintf("enableVlanFiltering on %s failed: %s",
+				bridgeName, err)
+			log.Errorf(errStr)
+		}
+	}
+}
+
+func setupVlans(vifList []types.VifInfo) {
```

Comment:
Can this be called more than once? It looks like enableVlanFiltering can handle that, but this function would add the same vlans again it it was called. Is that a possibility?

Reply:
This function is called from doActivateTail. And doActivateTail is called from doActivate, maybeRetryBoot function. setupVlans is called at the end when the domU boot is successfull. So, it will not be called more than once.